### PR TITLE
[Hexagon] Add AoT capability to Hexagon launcher

### PR DESCRIPTION
**Pull request by kparzysz-quic**

### CODE REVIEW

File: apps/hexagon_launcher/launcher_hexagon.cc [hunk 1]
```diff
@@ -64,7 +64,22 @@ AEEResult __QAIC_HEADER(launcher_rpc_load)(remote_handle64 handle, const char* m
   }
 
   tvm::runtime::Module module = load_module(module_path);
-  tvm::runtime::Module executor = create_graph_executor(graph_json, module, Model::device());
+  std::string module_type = module->type_key();
+  tvm::runtime::Module executor;
+  if (module_type == "AotExecutorFactory") {
+    executor = create_aot_executor(module, Model::external());
+  } else if (module_type == "library") {
+    // We're not expecting "GraphExecutorFactory" here.
```
Comment (csullivan) [inline apps/hexagon_launcher/launcher_hexagon.cc:72]: What's the root cause of the difference in module type? I would expect executor factories in both cases

Comment (kparzysz-quic) [reply apps/hexagon_launcher/launcher_hexagon.cc:72]: We never saved the factory for graph executor---instead we saved the library module and created the executor by hand.  This is just a reflection of how the original code worked.

Comment (csullivan) [reply apps/hexagon_launcher/launcher_hexagon.cc:72]: I see, other than it isn't necessary, was there a reason to not save the factory in the case of the graph executor?

Comment (kparzysz-quic) [reply apps/hexagon_launcher/launcher_hexagon.cc:72]: I've never tried saving it.  I think we may need to do `export_library` for that, which only works with target `llvm`.  The original idea was to strip all the "factories" and just save the executable code for simplicity.

**Review by csullivan — APPROVED** [review]
LGTM

**Pull request merged**